deploy/playbooks/roles/postgresql/tasks/users.yml

```diff
@@ -1,7 +1,13 @@
 ---
 
 - name: generate pseudo-random password for the postgres replication user
-  shell: python -c "exec 'import os; print os.urandom(30).encode(\'base64\')[:${length}]'"
+  local_action: shell python -c "exec 'import os; print os.urandom(30).encode(\'base64\')[:${length}]'" | tee {{ playbook_dir }}/replication_user_password
+  register: generated_replication_user_password
+  changed_when: false
+  when: postgres_replication_user_password is not defined
```

Comment:
I think you can use ``creates`` with the ``shell`` module so this command doesn't run if the ``replication_user_password`` file already exists.